connector/src/test/java/com/google/cloud/hive/bigquery/connector/IntegrationTests.java

```diff
@@ -157,4 +161,148 @@ public void testUnsupportedTypes() {
     }
     tearDown();
   }
+
+  // ---------------------------------------------------------------------------------------------------
+
+  /** Check that reading an empty BQ table actually returns 0 results. */
+  public void readEmptyTable(String engine, String readDataFormat) {
+    runBqQuery(BIGQUERY_TEST_TABLE_CREATE_QUERY);
+    initHive(engine, readDataFormat);
+    hive.execute(HIVE_TEST_TABLE_CREATE_QUERY);
+    List<Object[]> rows = hive.executeStatement(String.format("SELECT * FROM %s", TEST_TABLE_NAME));
+    assertEquals(0, rows.size());
+  }
+
+  @Test
+  public void testReadEmptyTable() {
```

Comment:
Please split to two methods - testReadEmptyTable_mr(), testReadEmptyTable_tez(), so when there is a failure we will know if it's in a single engine or both (and same for other relevant methods)

Reply:
So that's what I had originally but that exploded into a huge number of methods. That's because it's useful to test for multiple execution engine, multiple read data format, and multiple write method. The number of methods got a bit unwieldy, which is why I used nested loops. I wonder if maybe there's a way to clearly display the loop values when there's a failure?

Reply:
Ok, I see there's also a `@ParameterizedTest` decorator combined with `@ValueSource`. Would that work?

Also, would it be ok to change that once all the PRs have been merged to avoid too many rebase conflicts?